app/api/auth.py

```diff
@@ -80,9 +80,52 @@ def login():
                response_all_user_details = requests.post(url=URL, json=payload_all_user_details)
                result = response_all_user_details.json()
                data = result['data']
-               user = mongo.db.users.insert({
-                   "profile": data
-               })
+
+               for user in data:
+                  username = user['username']
+                  id  = user['id']
+                  name = user['name']
+                  user_Id = user['user_Id']
+                  status = user['status']
+                  jobtitle = user['jobtitle']
+                  dob = user['dob']
+                  gender = user['gender']
+                  work_email = user['work_email']
+                  slack_id = user['slack_id']
+                  user = mongo.db.users.count({
+                      "username": user['username']})
+                  if user > 0:
+                      user = mongo.db.users.update({
+                          "username": username

```

Comment:
there is a feature in mongo. when doing update we can set insert = true use that. all above will happen with a single query

Reply:
ok sir